Remove pointless null checks that are extremely unlikely to fail.  (Bug 156716)

diff --git a/layout/style/nsStyleSet.cpp b/layout/style/nsStyleSet.cpp
--- a/layout/style/nsStyleSet.cpp
+++ b/layout/style/nsStyleSet.cpp
@@ -638,82 +638,57 @@ nsStyleSet::ResolveStyleFor(nsIContent* 
   nsStyleContext*  result = nsnull;
   nsPresContext* presContext = PresContext();
 
   NS_ASSERTION(aContent, "must have content");
   NS_ASSERTION(aContent->IsNodeOfType(nsINode::eELEMENT),
                "content must be element");
 
   if (aContent && presContext) {
-    if (mRuleProcessors[eAgentSheet]        ||
-        mRuleProcessors[ePresHintSheet]     ||
-        mRuleProcessors[eUserSheet]         ||
-        mRuleProcessors[eHTMLPresHintSheet] ||
-        mRuleProcessors[eDocSheet]          ||
-        mRuleProcessors[eStyleAttrSheet]    ||
-        mRuleProcessors[eOverrideSheet]) {
-      ElementRuleProcessorData data(presContext, aContent, mRuleWalker);
-      FileRules(EnumRulesMatching, &data);
-      result = GetContext(presContext, aParentContext, nsnull).get();
+    ElementRuleProcessorData data(presContext, aContent, mRuleWalker);
+    FileRules(EnumRulesMatching, &data);
+    result = GetContext(presContext, aParentContext, nsnull).get();
 
-      // Now reset the walker back to the root of the tree.
-      mRuleWalker->Reset();
-    }
+    // Now reset the walker back to the root of the tree.
+    mRuleWalker->Reset();
   }
 
   return result;
 }
 
 already_AddRefed<nsStyleContext>
 nsStyleSet::ResolveStyleForRules(nsStyleContext* aParentContext, const nsCOMArray<nsIStyleRule> &rules)
 {
   NS_ENSURE_FALSE(mInShutdown, nsnull);
   nsStyleContext* result = nsnull;
   nsPresContext *presContext = PresContext();
 
   if (presContext) {
-    if (mRuleProcessors[eAgentSheet]        ||
-        mRuleProcessors[ePresHintSheet]     ||
-        mRuleProcessors[eUserSheet]         ||
-        mRuleProcessors[eHTMLPresHintSheet] ||
-        mRuleProcessors[eDocSheet]          ||
-        mRuleProcessors[eStyleAttrSheet]    ||
-        mRuleProcessors[eOverrideSheet]) {
-      
-      mRuleWalker->SetLevel(eDocSheet, PR_FALSE);
-      for (PRInt32 i = 0; i < rules.Count(); i++) {
-        mRuleWalker->Forward(rules.ObjectAt(i));
-      }
-      result = GetContext(presContext, aParentContext, nsnull).get();
+    mRuleWalker->SetLevel(eDocSheet, PR_FALSE);
+    for (PRInt32 i = 0; i < rules.Count(); i++) {
+      mRuleWalker->Forward(rules.ObjectAt(i));
+    }
+    result = GetContext(presContext, aParentContext, nsnull).get();
 
-      // Now reset the walker back to the root of the tree.
-      mRuleWalker->Reset();
-    }
+    // Now reset the walker back to the root of the tree.
+    mRuleWalker->Reset();
   }
   return result;
 }
 
 already_AddRefed<nsStyleContext>
 nsStyleSet::ResolveStyleForNonElement(nsStyleContext* aParentContext)
 {
   nsStyleContext* result = nsnull;
   nsPresContext *presContext = PresContext();
 
   if (presContext) {
-    if (mRuleProcessors[eAgentSheet]        ||
-        mRuleProcessors[ePresHintSheet]     ||
-        mRuleProcessors[eUserSheet]         ||
-        mRuleProcessors[eHTMLPresHintSheet] ||
-        mRuleProcessors[eDocSheet]          ||
-        mRuleProcessors[eStyleAttrSheet]    ||
-        mRuleProcessors[eOverrideSheet]) {
-      result = GetContext(presContext, aParentContext,
-                          nsCSSAnonBoxes::mozNonElement).get();
-      NS_ASSERTION(mRuleWalker->AtRoot(), "rule walker must be at root");
-    }
+    result = GetContext(presContext, aParentContext,
+                        nsCSSAnonBoxes::mozNonElement).get();
+    NS_ASSERTION(mRuleWalker->AtRoot(), "rule walker must be at root");
   }
 
   return result;
 }
 
 
 static PRBool
 EnumPseudoRulesMatching(nsIStyleRuleProcessor* aProcessor, void* aData)
@@ -743,32 +718,24 @@ nsStyleSet::ResolvePseudoStyleFor(nsICon
   NS_ASSERTION(aParentContent ||
                nsCSSAnonBoxes::IsAnonBox(aPseudoTag),
                "null content must correspond to anonymous box");
   NS_ASSERTION(nsCSSAnonBoxes::IsAnonBox(aPseudoTag) ||
                nsCSSPseudoElements::IsPseudoElement(aPseudoTag),
                "aPseudoTag must be pseudo-element or anonymous box");
 
   if (aPseudoTag && presContext) {
-    if (mRuleProcessors[eAgentSheet]        ||
-        mRuleProcessors[ePresHintSheet]     ||
-        mRuleProcessors[eUserSheet]         ||
-        mRuleProcessors[eHTMLPresHintSheet] ||
-        mRuleProcessors[eDocSheet]          ||
-        mRuleProcessors[eStyleAttrSheet]    ||
-        mRuleProcessors[eOverrideSheet]) {
-      PseudoRuleProcessorData data(presContext, aParentContent, aPseudoTag,
-                                   aComparator, mRuleWalker);
-      FileRules(EnumPseudoRulesMatching, &data);
+    PseudoRuleProcessorData data(presContext, aParentContent, aPseudoTag,
+                                 aComparator, mRuleWalker);
+    FileRules(EnumPseudoRulesMatching, &data);
 
-      result = GetContext(presContext, aParentContext, aPseudoTag).get();
+    result = GetContext(presContext, aParentContext, aPseudoTag).get();
 
-      // Now reset the walker back to the root of the tree.
-      mRuleWalker->Reset();
-    }
+    // Now reset the walker back to the root of the tree.
+    mRuleWalker->Reset();
   }
 
   return result;
 }
 
 already_AddRefed<nsStyleContext>
 nsStyleSet::ProbePseudoStyleFor(nsIContent* aParentContent,
                                 nsIAtom* aPseudoTag,
@@ -788,33 +755,25 @@ nsStyleSet::ProbePseudoStyleFor(nsIConte
   NS_ASSERTION(aParentContent ||
                nsCSSAnonBoxes::IsAnonBox(aPseudoTag),
                "null content must correspond to anonymous box");
   NS_ASSERTION(nsCSSAnonBoxes::IsAnonBox(aPseudoTag) ||
                nsCSSPseudoElements::IsPseudoElement(aPseudoTag),
                "aPseudoTag must be pseudo-element or anonymous box");
 
   if (aPseudoTag && presContext) {
-    if (mRuleProcessors[eAgentSheet]        ||
-        mRuleProcessors[ePresHintSheet]     ||
-        mRuleProcessors[eUserSheet]         ||
-        mRuleProcessors[eHTMLPresHintSheet] ||
-        mRuleProcessors[eDocSheet]          ||
-        mRuleProcessors[eStyleAttrSheet]    ||
-        mRuleProcessors[eOverrideSheet]) {
-      PseudoRuleProcessorData data(presContext, aParentContent, aPseudoTag,
-                                   nsnull, mRuleWalker);
-      FileRules(EnumPseudoRulesMatching, &data);
+    PseudoRuleProcessorData data(presContext, aParentContent, aPseudoTag,
+                                 nsnull, mRuleWalker);
+    FileRules(EnumPseudoRulesMatching, &data);
 
-      if (!mRuleWalker->AtRoot())
-        result = GetContext(presContext, aParentContext, aPseudoTag).get();
+    if (!mRuleWalker->AtRoot())
+      result = GetContext(presContext, aParentContext, aPseudoTag).get();
 
-      // Now reset the walker back to the root of the tree.
-      mRuleWalker->Reset();
-    }
+    // Now reset the walker back to the root of the tree.
+    mRuleWalker->Reset();
   }
 
   // For :before and :after pseudo-elements, having display: none or no
   // 'content' property is equivalent to not having the pseudo-element
   // at all.
   if (result &&
       (aPseudoTag == nsCSSPseudoElements::before ||
        aPseudoTag == nsCSSPseudoElements::after)) {
@@ -941,24 +900,17 @@ static PRBool SheetHasStatefulStyle(nsIS
 // Test if style is dependent on content state
 nsReStyleHint
 nsStyleSet::HasStateDependentStyle(nsPresContext* aPresContext,
                                    nsIContent*     aContent,
                                    PRInt32         aStateMask)
 {
   nsReStyleHint result = nsReStyleHint(0);
 
-  if (aContent->IsNodeOfType(nsINode::eELEMENT) &&
-      (mRuleProcessors[eAgentSheet]        ||
-       mRuleProcessors[ePresHintSheet]     ||
-       mRuleProcessors[eUserSheet]         ||
-       mRuleProcessors[eHTMLPresHintSheet] ||
-       mRuleProcessors[eDocSheet]          ||
-       mRuleProcessors[eStyleAttrSheet]    ||
-       mRuleProcessors[eOverrideSheet])) {
+  if (aContent->IsNodeOfType(nsINode::eELEMENT)) {
     StatefulData data(aPresContext, aContent, aStateMask);
     WalkRuleProcessors(SheetHasStatefulStyle, &data);
     result = data.mHint;
   }
 
   return result;
 }
 
@@ -988,24 +940,17 @@ nsStyleSet::HasAttributeDependentStyle(n
 nsStyleSet::HasAttributeDependentStyle(nsPresContext* aPresContext,
                                        nsIContent*    aContent,
                                        nsIAtom*       aAttribute,
                                        PRInt32        aModType,
                                        PRUint32       aStateMask)
 {
   nsReStyleHint result = nsReStyleHint(0);
 
-  if (aContent->IsNodeOfType(nsINode::eELEMENT) &&
-      (mRuleProcessors[eAgentSheet]        ||
-       mRuleProcessors[ePresHintSheet]     ||
-       mRuleProcessors[eUserSheet]         ||
-       mRuleProcessors[eHTMLPresHintSheet] ||
-       mRuleProcessors[eDocSheet]          ||
-       mRuleProcessors[eStyleAttrSheet]    ||
-       mRuleProcessors[eOverrideSheet])) {
+  if (aContent->IsNodeOfType(nsINode::eELEMENT)) {
     AttributeData data(aPresContext, aContent, aAttribute, aModType,
                        aStateMask);
     WalkRuleProcessors(SheetHasAttributeStyle, &data);
     result = data.mHint;
   }
 
   return result;
 }
